src/zookeeper.c

```diff
@@ -176,9 +176,11 @@ static int zookeeper_read(void) {
   char *save_ptr;
   char *line;
   char *fields[2];
+  long followers = 0;
 
   if (zookeeper_query(buf, sizeof(buf)) < 0) {
-    return -1;
+    zookeeper_submit_gauge("count", "quorum", -1);
+    return 0;
```

Comment:
I don't entirely understand the implications of this change.  Does returning 0 here seem like it would cause any issues?

Reply:
If this returns non-zero, collectd treats this as "callback failed" and will start doing an exponential back-off, i.e. the function will be called less and less frequently until it recovers. If you return zero here, the function will be called periodically, even if `zookeeper_query()` fails. That may waste some resources, but since the plugin currently re-establishes a connection every time it reads metrics, I don't think that's an issue.

Keep in mind, though, that the connect timeout may be longer than the report interval, so you may get metrics less frequently than you'd expect in certain network condtions.

Reply:
I think the long timeout and additional resources are acceptable, as long as it isn't causing problems elsewhere in collectd.